pkg/manager/member/tidb_init_manager.go

```diff
@@ -169,7 +169,7 @@ func (m *tidbInitManager) syncTiDBInitConfigMap(ti *v1alpha1.TidbInitializer) er
 	}
 
 	tlsClientEnabled := false
-	if tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() {
+	if tc.Spec.TiDB != nil && tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() {
```

Comment:
I think we should list the tc in the `sync` function https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tidb_init_manager.go#L66, print log and return nil if `spec.tidb=nil`.